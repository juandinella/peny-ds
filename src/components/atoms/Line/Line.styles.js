import { css } from 'styled-components'

export default css`
  width: 100%;
  border-top: 1px;
  background: none;
  margin: 0;

  ${(props) =>
    props.color &&
    css`
      border-color: var(--color-${(props) => props.color});
    `}

  ${(props) =>
    props.type &&
    css`
      border-style: none;
      border-top-style: ${(props) => props.type};
    `}
`
