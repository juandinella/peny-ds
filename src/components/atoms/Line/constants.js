export const options = {
  colors: [
    'neutral-00',
    'neutral-50',
    'neutral-100',
    'neutral-200',
    'neutral-300',
    'neutral-400',
    'neutral-500',
    'neutral-600',
    'neutral-700',
    'neutral-800',
    'neutral-900',
    'primary-main',
    'secondary-main',
    'alert-error',
    'alert-success',
    'alert-warning',
  ],
  types: ['solid', 'dashed'],
}
