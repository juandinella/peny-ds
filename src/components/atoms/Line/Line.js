import React from 'react'
import PropTypes from 'prop-types'
import styles from './Line.styles'
import { options } from './constants'
import Skeleton from '../Skeleton'
import Utils from '../../../utils'

/**
 * Just a line
 */

const Line = ({ id, type, color, shouldRender }) => {
  return (
    <Skeleton
      id={`${id}_skeleton`}
      show={!Utils.emptyValue(shouldRender) && !shouldRender()}
    >
      <hr css={styles} id={id} data-testid={id} type={type} color={color}></hr>
    </Skeleton>
  )
}

Line.displayName = 'Line'

Line.propTypes = {
  /** Lne's id */
  id: PropTypes.string,
  /** Type of the line */
  type: PropTypes.oneOf(options.types),
  /** Color of the line */
  color: PropTypes.oneOf(options.colors),
  /**
   * Function that decides if the component should be rendered.
   *
   * @return {boolean} false if component should not be rendered and show skeleton instead. Component renders otherwise.
   */
  shouldRender: PropTypes.func,
}

Line.defaultProps = {
  color: 'neutral-900',
  type: 'solid',
}

export default Line
