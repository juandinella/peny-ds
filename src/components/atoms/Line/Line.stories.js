import Line from './Line'
import { options } from './constants'

import {
  getTemplate,
  getListTemplate,
  getOptionsArgTypes,
} from '../../../helpers/storybook'

const Template = getTemplate(Line)
const ListTemplate = getListTemplate(Line)

export default {
  title: 'Atoms/Line',
  component: Line,
  args: {},
  argTypes: {
    color: getOptionsArgTypes(options.colors),
    type: getOptionsArgTypes(options.types),
  },
}

export const Default = Template.bind({})

export const Colors = ListTemplate.bind({})
Colors.args = { items: options.colors.map((color) => ({ color })) }

export const Types = ListTemplate.bind({})
Types.args = { items: options.types.map((type) => ({ type })) }

export const Skeleton = Template.bind({})
Skeleton.args = { shouldRender: () => false }
