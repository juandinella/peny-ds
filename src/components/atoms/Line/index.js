import Line from './Line'

export default Line
export { options } from './constants'
