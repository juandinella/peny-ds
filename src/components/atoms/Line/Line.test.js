import React from 'react'

import { render, cleanup } from '@testing-library/react'

import Line from './Line'

import initStoryshots from '@storybook/addon-storyshots'

initStoryshots({
  storyKindRegex: /^Atoms\/Line$/,
})

const componentProps = {
  id: 'unique_id',
}

afterEach(cleanup)

describe('<Line />', () => {
  test('should render component correctly', () => {
    const { queryByTestId } = render(<Line {...componentProps} />)

    const line = queryByTestId(componentProps.id)

    expect(line).toBeTruthy()
  })

  test('should render skeleton', () => {
    const { queryByTestId } = render(
      <Line {...componentProps} shouldRender={() => false} />
    )

    expect(queryByTestId(`${componentProps.id}_skeleton`)).toBeTruthy()
  })
})
