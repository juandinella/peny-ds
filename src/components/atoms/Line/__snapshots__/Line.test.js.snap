// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Line Colors 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <hr
    className="sc-atln6s-0 fKGCGy"
    color="neutral-120"
    type="solid"
  />
  <hr
    className="sc-atln6s-0 eJqAEI"
    color="neutral-100"
    type="solid"
  />
  <hr
    className="sc-atln6s-0 lkkBLp"
    color="neutral-80"
    type="solid"
  />
  <hr
    className="sc-atln6s-0 dpFhlH"
    color="neutral-60"
    type="solid"
  />
  <hr
    className="sc-atln6s-0 dddXwh"
    color="neutral-40"
    type="solid"
  />
  <hr
    className="sc-atln6s-0 dSxLjL"
    color="neutral-20"
    type="solid"
  />
  <hr
    className="sc-atln6s-0 dFWBul"
    color="neutral-00"
    type="solid"
  />
  <hr
    className="sc-atln6s-0 GPOIX"
    color="primary-main"
    type="solid"
  />
  <hr
    className="sc-atln6s-0 hziGOi"
    color="alert-error"
    type="solid"
  />
  <hr
    className="sc-atln6s-0 gKonp"
    color="alert-success"
    type="solid"
  />
  <hr
    className="sc-atln6s-0 fRXSho"
    color="alert-warning"
    type="solid"
  />
</div>
`;

exports[`Storyshots Atoms/Line Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <hr
    className="sc-atln6s-0 fKGCGy"
    color="neutral-120"
    type="solid"
  />
</div>
`;

exports[`Storyshots Atoms/Line Skeleton 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-1osow0y-0 dpAisn"
    data-testid="undefined_skeleton"
    id="undefined_skeleton"
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <hr
      className="sc-atln6s-0 fKGCGy"
      color="neutral-120"
      type="solid"
    />
  </div>
</div>
`;

exports[`Storyshots Atoms/Line Types 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <hr
    className="sc-atln6s-0 fKGCGy"
    color="neutral-120"
    type="solid"
  />
  <hr
    className="sc-atln6s-0 KtRBU"
    color="neutral-120"
    type="dashed"
  />
</div>
`;
